support-frontend/assets/helpers/redux/checkout/product/selectors/productPrice.ts

```diff
@@ -0,0 +1,99 @@
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { productOptionIfDigiAddOnChanged } from 'helpers/productPrice/productOptions';
+import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import {
+	finalPrice,
+	getProductPrice,
+} from 'helpers/productPrice/productPrices';
+import { paperProductTypes } from 'helpers/productPrice/subscriptions';
+import type { SubscriptionsState } from 'helpers/redux/subscriptionsStore';
+import type { GuardianProduct, ProductState } from '../state';
+
+function canDeterminePaperPrice(productState: ProductState): boolean {
+	const hasFulfilment = productState.fulfilmentOption !== 'NoFulfilmentOptions';
+	const hasProductOption = productState.productOption !== 'NoProductOptions';
+	return hasFulfilment && hasProductOption;
+}
+
+const requiredFieldsForProduct: Record<
+	GuardianProduct,
+	(productState: ProductState) => boolean
+> = {
+	DigitalPack: () => true,
+	GuardianWeekly: (productState: ProductState) =>
+		productState.fulfilmentOption !== 'NoFulfilmentOptions',
+	Paper: canDeterminePaperPrice,
+	PaperAndDigital: canDeterminePaperPrice,
+	NoProduct: () => false,
+	DailyEdition: () => false,
+	PremiumTier: () => false,
+	ONE_OFF: () => false,
+	MONTHLY: () => false,
+	ANNUAL: () => false,
+};
+
+const getDefaultPriceObject = (currency: IsoCurrency): ProductPrice => ({
+	price: 0,
+	currency,
+	fixedTerm: false,
+});
+
+export function selectPriceForProduct(state: SubscriptionsState): ProductPrice {
+	const { common, page } = state;
+	const { product } = page.checkoutForm;
+	if (requiredFieldsForProduct[product.productType](product)) {
+		const selectedProductPrice = getProductPrice(
+			product.productPrices,
+			common.internationalisation.countryId,
+			product.billingPeriod,
+			product.fulfilmentOption,
+			product.productOption,
+		);
+		return selectedProductPrice;
+	}
+	return getDefaultPriceObject(common.internationalisation.currencyId);
```

Comment:
Is it safe to return a default price here? Is there a risk that we end up charging 0 for a subscription in which case it might be better to throw and error?

Reply:
You're right, it's highly unlikely that we'd ever hit this but throwing is a better choice.